### PR TITLE
Support for building static osquery executable

### LINKED ISSUE #2198: Determine the requirements of delivering a static windows binary
Similarly to how we compile binaries on OS X and Linux, let's explore how we can compile the static libraries of our dependencies into the osqueryi.exe and osqueryd.exe binaries on Windows. Once we have a better story around how accomplishable this is, we can do some more extensive project management to figure out when/how we should deliver this.

### PR DESCRIPTION
**Pull request by yying**
This will allow static osquery executables to be built. The provision script grabs dependencies from a new URL that contains the static library version of dependencies.

**NOTE**: <del>This PR depends on https://github.com/osquery/third-party/pull/12 and updating `third-party/` submodule to point to the latest commit</del>

Closes #2198

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by PoppySeedPlehzr** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit a1cd2270a991c3448c21877a99eb942f4a5b6776 (Job results: [3106](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3106/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit a1cd2270a991c3448c21877a99eb942f4a5b6776 (Job results: [124](https://jenkins.osquery.io/job/osqueryPullRequestBuild/124/)) failed one or more tests (**Windows**).

File: osquery/CMakeLists.txt [hunk 1]
```diff
@@ -62,7 +62,7 @@ if(LINUX OR FREEBSD)
   ADD_OSQUERY_LINK_CORE("libc")
 endif()
 
-ADD_OSQUERY_LINK_CORE("glog")
+ADD_OSQUERY_LINK_CORE("libglog_static")
```
Comment (theopolis) [inline osquery/CMakeLists.txt]: Maybe drop the "lib" prefix?

**Comment by osqueryer** [comment]
:-1: The commit 79747b15d02fd31462eca211ee7bab30d5efff6b (Job results: [3112](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3112/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 79747b15d02fd31462eca211ee7bab30d5efff6b (Job results: [130](https://jenkins.osquery.io/job/osqueryPullRequestBuild/130/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 44f3d4a2b2c0d1cfdf043b5b1adee9abfdb23f37 (Job results: [3113](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3113/)) failed one or more tests (**OS X/Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 44f3d4a2b2c0d1cfdf043b5b1adee9abfdb23f37 (Job results: [131](https://jenkins.osquery.io/job/osqueryPullRequestBuild/131/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 61ba0e45b212738bd7ddb0b100828caa4b5152dd (Job results: [3119](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3119/)) failed one or more tests (**OS X/Linux**).

**Pull request merged**